docs/user/customization.md

```diff
@@ -64,4 +64,106 @@ An example `worker.ign` is shown below. It has been modified to increase the HTT
 }
 ```
 
+## Install time customization for machine configuration
+
+In most cases, user applications should be run on the cluster via Kubernetes workload objects (e.g. daemonset, deployment, etc). For example, if a user wants to run a logging agent on all hosts, in general we would want to steer them towards deploying it as a daemonset object. However, there may be some cases where these workloads need to be executed prior to the node joining the Kubernetes cluster. A specific example that was given from a customer call was that due to a compliance mandate "they must run auditing the second the operating system comes up". In this case, the admin would need the ability to define a custom section of Ignition config (containing the systemd unit that runs the auditing container).
+
+The configuration of machines in OpenShift is controlled using [MachineConfigs][machine-config] and what configuration is applied to a machine in the OpenShift cluster is based on the [MachineConfigPools][machine-config-pool]. To allow customization of machine configuration which is not possible as Day 2 operation, the installer allows users to bring their own custom `machineconfig` objects.
+
+1. `openshift-install create manifests --dir $INSTALL_DIR`
+
+2. Copy files with `machineconfig` objects to `$INSTALL_DIR/openshift/`
+
+3. `openshift-install create cluster --dir $INSTALL_DIR`
+
+These custom `machineconfig` objects are black boxes to the installer and the installer only plays the role of `oc create -f <custom-machine-config-object>` early enough into cluster bootstrap to make sure the configuration is used by the [MachineConfigOperator][machine-config-operator].
+
+For example, all control plane nodes register with taint `node-role.kubernetes.io/master=:NoSchedule` making them unschedulable by most normal workloads. An imaginary installation that requires the control plane to boot without that taint can push a custom `machineconfig` with `kubelet.service` that doesn't include the taint.
+
+On Libvirt:
+
+1. Run `manifests` target to create all the manifests.
+
+```console
+ls -l no-taint-cluster/**
+```
+
+2. Create a `machineconfig` that includes `kubelet.service` that has no taints.
+
+```console
+$ cat > no-taint-cluster/openshift/99-master-kubelet-no-taint.yaml <<EOF
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: master
+  name: 02-master-kubelet
+spec:
+  config:
+    ignition:
+      config: {}
+      security:
+        tls: {}
+      timeouts: {}
+      version: 2.2.0
```

Comment:
Link over [here][1] with a big caveat saying `create manifests` is explicitly not covered by our compatibility commitment?  I don't want folks to be surprised when we bump to Ignition 3.0 or reshuffle our asset tree and break this.

[1]: https://github.com/openshift/installer/blob/df201360badf4780f8b1fe96d0467576ee73fa1a/docs/user/versioning.md